src/index.ts

```diff
@@ -23,19 +23,53 @@ const defaultApps = [
   require('./plugins/default')
 ]
 
-export const createProbot = (options: Options) => {
-  options.webhookPath = options.webhookPath || '/'
-  options.secret = options.secret || 'development'
+export class Probot {
+  public server: express.Application
+  public webhook: any
+  public logger: Logger
+
+  private options: Options
+  private robots: Robot[]
+  private app: () => string
+
+  constructor(options: Options) {
+    options.webhookPath = options.webhookPath || '/'
+    options.secret = options.secret || 'development'
+
+    this.options = options
+    this.logger = logger
+
+    this.webhook = new Webhooks({path: options.webhookPath, secret: options.secret})
+
+    this.app = createApp({
+      id: options.id,
+      cert: options.cert
+    })
+
+    this.server = createServer({webhook: this.webhook.middleware, logger})
 
-  const webhook = new Webhooks({path: options.webhookPath, secret: options.secret})
-  const app = createApp({
-    id: options.id,
-    cert: options.cert
-  })
-  const server: express.Application = createServer({webhook: webhook.middleware, logger})
+    // Log all received webhooks
+    this.webhook.on('*', (event: any) => {
+      const webhookEvent = {
```

Comment:
I thought of a better (or at least different) way to do this (picking and deleting a field). I can add that now or in a followup:

```js
const { name, ...webhookEvent } = { ...event, event: event.name }
```